Meziantou.FluentAssertionsAnalyzers.Tests/XUnitToFluentAssertionsAnalyzerUnitTests.cs

```diff
@@ -510,6 +510,123 @@ public void MyTest()
         obj.Should().Be(expected);
     }
 }
+");
+    }
+    [Fact]
+    public Task Assert_Equal_T_T_IEqualityComparerOfT()
+    {
+        return Assert(@"
+using Xunit;
+using FluentAssertions;
+using System.Collections.Generic;
+
+public class TestObject
+{
+	private sealed class NameEqualityComparer : IEqualityComparer<TestObject>
+	{
+		public bool Equals(TestObject x, TestObject y)
+		{
+			if (ReferenceEquals(x, y))
+			{
+				return true;
+			}
+
+			if (ReferenceEquals(x, null))
+			{
+				return false;
+			}
+
+			if (ReferenceEquals(y, null))
+			{
+				return false;
+			}
+
+			if (x.GetType() != y.GetType())
+			{
+				return false;
+			}
+
+			return x.Name == y.Name;
+		}
+
+		public int GetHashCode(TestObject obj)
+		{
+			return obj.Name != null
+				? obj.Name.GetHashCode()
+				: 0;
+		}
+	}

```

Comment:
```suggestion
		public bool Equals(TestObject x, TestObject y)
The implementation of this object doesn't matter. You can remove it.

```suggestion
		public bool Equals(TestObject x, TestObject y) => throw null;
```